pkg/controller/cronjob/utils.go

```diff
@@ -63,17 +63,23 @@ func deleteFromActiveList(sj *batchv2alpha1.CronJob, uid types.UID) {
 
 // getParentUIDFromJob extracts UID of job's parent and whether it was found
 func getParentUIDFromJob(j batchv1.Job) (types.UID, bool) {
-	creatorRefJson, found := j.ObjectMeta.Annotations[v1.CreatedByAnnotation]
-	if !found {
-		glog.V(4).Infof("Job with no created-by annotation, name %s namespace %s", j.Name, j.Namespace)
+	creatorRefJson, err := makeCreatedByRefJson(&j)
```

Comment:
you are definitely right about this. i found several ways to obtain the same result, and chose a poor one. it is a good lesson for me - i will change my codes accordingly.